src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileReader.kt

```diff
@@ -107,4 +111,25 @@ class CsvFileReader internal constructor(
         }
         return null
     }
+
+    /**
+     * deduplicate headers based on occurrence by appending "_<NUM>"
+     * Ex: [a,b,b,c,a] => [a,b,b_2,c,a_2]
+     *
+     * @return return headers as List<String>.
+     *
+     */
+    private fun deduplicateHeaders(headers: List<String>): List<String> {
+        val occurrences = headers.toSet().associateWith { 1 }.toMutableMap()
+        return headers.map { header ->
+            occurrences[header]?.let { occurred ->
+                when {
+                    occurred > 1 -> "${header}_$occurred"
+                    else -> header
+                }.also {
+                    occurrences[header] = occurred + 1
+                }
+            } ?: throw RuntimeException("a")
```

Comment:
Be aware that this is case insensitive. So you still can have (`Example`,`exAmple`, `example`). As of now this purposely doesn't catch ambiguous duplicates.

Reply:
You said `case insensitive`, but is that misuse of `case sensitive`?

Reply:
Yes, fixed it in the original comment.